### PR TITLE
Update references to Suricata website - v1

### PR DESCRIPTION
**Pull request by jufajardini**
Many places were still referencing the old Suricata page. 
Used git grep with replace to update them. Checked that new links work. 
Left old references when they were only documentation examples (for
output or unittests, for instance).

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/4915

Describe changes:
- replaced references to suricata-ids.org with suricata.io over several different pages, files and docs.
- fixed typos in userguide page about suricata.yaml

### CODE REVIEW

**Comment by jufajardini** [comment]
Added a redmine ticket: https://redmine.openinfosecfoundation.org/issues/4915

File: doc/AUTHORS [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Team:
-https://suricata-ids.org/about/team/
+https://suricata.io/about/team/
 
 All contributors:
 https://www.ohloh.net/p/suricata-engine/contributors/summary
```
Comment (jasonish) [inline doc/AUTHORS:5]: Ohloh appears to be dead? I know its not part of this PR but falls into URL cleanup.

**Review by jasonish — APPROVED** [review]
Looks correct to me.

File: doc/AUTHORS (hunk 1 above)
Comment (jufajardini) [reply doc/AUTHORS:5]: True, Shivani had mentioned that a while ago, as well... this works: https://www.openhub.net/p/suricata-engine/contributors/summary
Would it be interesting if I replaced ohloh with that one?

**Comment by jufajardini** [comment]
Added ohloh change into next PR https://github.com/OISF/suricata/pull/6739
